src/utils/AuthApi.js

```diff
@@ -0,0 +1,39 @@
+import {BASE_URL} from "./Constants";
+
+export const register = (email, password, name) => {
+  return fetch(`${BASE_URL}/signup`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({email, password, name})
+  })
```

Comment:
You need to check every response from the server in `AuthApi.js` the same way  you do in `MainApi` with `_checkRes` to invoke an error if the response is not `ok`.